cfdm/read_write/netcdf/netcdfwrite.py

```diff
@@ -4723,23 +4901,18 @@ def write(
         # ------------------------------------------------------------
         # Open the output netCDF file
         # ------------------------------------------------------------
-        filename = os.path.expanduser(os.path.expandvars(filename))
-        if os.path.isfile(filename):
+        if os.path.isfile(filename) and mode == "w":
```

Comment:
I think the logic here is a bit awry, as _not_ setting overwrite=False when the file does not exist causes problems in `def file_open`

Replacing this line, and the following two if blocks, with
```python
        if os.path.isfile(filename):
            if mode == "w" and not overwrite:
                raise IOError(
                    "Can't write with mode {mode!r} to existing file "
                    f"{os.path.abspath(filename)} unless overwrite=True"
                )

            if not os.access(filename, os.W_OK):
                raise IOError(
                    "Can't write to existing file "
                    f"{os.path.abspath(filename)} without permission"
                )
        else:
            g["overwrite"] = False
```
should do it ...

Reply:
Ah, good spot and thanks for the suggested fix, I'll give it a try in the next commit and report back.

Reply:
The suggestion works perfectly, thanks, and has been added in one of the new commits I will push shortly.